ts/models/conversation.ts

```diff
@@ -1195,15 +1196,50 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
     }
   }
 
+  /**
+   * Mark everything as read efficiently if possible.
+   *
+   * For convos with a expiration timer enable, start the timer as of no.
+   * Send read receipt if needed.
+   */
+  public async markAllAsRead() {
+    if (this.isOpenGroupV2()) {
+      // for opengroups, we batch everything as there is no expiration timer to take care (and potentially a lot of messages)
+
+      await Data.markAllAsReadByConversationNoExpiration(this.id);
+      this.set({ mentionedUs: false, unreadCount: 0 });
+
+      await this.commit();
+      return;
+    }
+
+    // if the conversation has no expiration timer, we can also batch everything, but we also need to send read receipts potentially
+    // so we grab them from the db
+    if (!this.get('expireTimer')) {
+      const allReadMessages = await Data.markAllAsReadByConversationNoExpiration(this.id);
+      this.set({ mentionedUs: false, unreadCount: 0 });
+      await this.commit();
+      if (allReadMessages.length) {
+        await this.sendReadReceiptsIfNeeded(uniq(allReadMessages));
+      }
+      return;
+    }
+
+    await this.markReadBouncy(Date.now());
+  }
+
   // tslint:disable-next-line: cyclomatic-complexity
-  public async markReadBouncy(newestUnreadDate: number, providedOptions: any = {}) {
+  public async markReadBouncy(
+    newestUnreadDate: number,
+    providedOptions: { sendReadReceipts?: boolean; readAt?: number } = {}
+  ) {
     const lastReadTimestamp = this.lastReadTimestamp;
     if (newestUnreadDate < lastReadTimestamp) {
       return;
     }
 
-    const options = providedOptions || {};
-    defaults(options, { sendReadReceipts: true });
+    const defaultedReadAt = providedOptions?.readAt || Date.now();
```

Comment:
defaulted normally means failed. Do you mean default?

Reply:
fixed